iio-sensor-proxy denial for sys_admin

rfrohl opened the pull request:
dontaudit:
type=AVC msg=audit(..): avc:  denied  { sys_admin } for  pid=6002 comm="iio-sensor-prox" capability=21  scontext=system_u:system_r:iiosensorproxy_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0

kernel_dgram_send(iiosensorproxy_t):
type=AVC msg=audit(..): avc:  denied  { sendto } for  pid=6002 comm="iio-sensor-prox" path="/systemd/journal/socket" scontext=system_u:system_r:iiosensorproxy_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=unix_dgram_socket permissive=0

Fixes: bsc#1244577


*EDIT:* We were missing d37dd6ab, but I still get the `sendto` AVC after retesting with that change included. I am therefor not sure if  `kernel_stream_connect` is needed or could be dropped.  [Reporter confirmed](https://bugzilla.suse.com/show_bug.cgi?id=1244577#c6) that sys_admin was not needed, so therefor dontaudit.

rfrohl commented:
found a late issue, need to confirm.

rfrohl commented:
Let me know if I should clean up ` kernel_stream_connect`. Happy to help if safes a bit of time. 

details, from test host with both commits:
> $ sesearch -s iiosensorproxy_t -t kernel_t -A -dt
> allow domain kernel_t:system module_request; [ domain_kernel_load_modules ]:True
> allow iiosensorproxy_t kernel_t:unix_dgram_socket sendto;
> allow iiosensorproxy_t kernel_t:unix_stream_socket { connectto getattr };

zpytela commented:
@rfrohl do you happen to know which syscall triggers the sys_admin capability?

rfrohl commented:
I only tested with dontaudit and got confirmation that it works, so I do not know from the top of my head. Is there an easy way to figure it out ? I still have a test system ready.

zpytela commented:
What is the kernel version? Can you check the same scenario with <=6.14.6?

https://bugzilla.redhat.com/show_bug.cgi?id=2369326

rfrohl commented:
6.15.2 ([full auditing](https://fedoraproject.org/wiki/SELinux/Debugging#Enable_full_auditing) enabled, with 'semodule -DB' and both fixes):

```
----                                                                                                                                                                         
type=PROCTITLE msg=audit(06/18/25 13:11:41.352:143) : proctitle=/usr/libexec/iio-sensor-proxy                                                                                
type=SYSCALL msg=audit(06/18/25 13:11:41.352:143) : arch=x86_64 syscall=setsockopt success=no exit=EPERM(Operation not permitted) a0=0x7 a1=SOL_SOCKET a2=SO_RCVBUFFORCE a3=0
x7ffde2d183b4 items=0 ppid=1 pid=1451 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=iio-sensor-prox ex
e=/usr/libexec/iio-sensor-proxy subj=system_u:system_r:iiosensorproxy_t:s0 key=(null)                                                                                        
type=AVC msg=audit(06/18/25 13:11:41.352:143) : avc:  denied  { net_admin } for  pid=1451 comm=iio-sensor-prox capability=net_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0                                                                                         
----                                                                                                                                                                         
type=PROCTITLE msg=audit(06/18/25 13:11:41.352:144) : proctitle=/usr/libexec/iio-sensor-proxy                                                                                
type=SYSCALL msg=audit(06/18/25 13:11:41.352:144) : arch=x86_64 syscall=setsockopt success=yes exit=0 a0=0x7 a1=SOL_SOCKET a2=SO_ATTACH_FILTER a3=0x7ffde2d175e0 items=0 ppid
=1 pid=1451 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=iio-sensor-prox exe=/usr/libexec/iio-sensor-
proxy subj=system_u:system_r:iiosensorproxy_t:s0 key=(null)                                                                                                                  
type=AVC msg=audit(06/18/25 13:11:41.352:144) : avc:  denied  { sys_admin } for  pid=1451 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0                                                                                         
type=AVC msg=audit(06/18/25 13:11:41.352:144) : avc:  denied  { sys_admin } for  pid=1451 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0                                                                                         
type=AVC msg=audit(06/18/25 13:11:41.352:144) : avc:  denied  { sys_admin } for  pid=1451 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0                                                                                         
type=AVC msg=audit(06/18/25 13:11:41.352:144) : avc:  denied  { sys_admin } for  pid=1451 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
type=AVC msg=audit(06/18/25 13:11:41.352:144) : avc:  denied  { sys_admin } for  pid=1451 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0                                                                                         
----
```


6.12.33 ([full auditing](https://fedoraproject.org/wiki/SELinux/Debugging#Enable_full_auditing) enabled , with 'semodule -DB' and both fixes):
```
----     
type=PROCTITLE msg=audit(06/18/25 13:18:47.947:286) : proctitle=/usr/libexec/iio-sensor-proxy                                                                                
type=SYSCALL msg=audit(06/18/25 13:18:47.947:286) : arch=x86_64 syscall=setsockopt success=no exit=EPERM(Operation not permitted) a0=0x7 a1=SOL_SOCKET a2=SO_RCVBUFFORCE a3=0
x7ffe02a89704 items=0 ppid=1 pid=1437 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=iio-sensor-prox ex
e=/usr/libexec/iio-sensor-proxy subj=system_u:system_r:iiosensorproxy_t:s0 key=(null)                                                                                        
type=AVC msg=audit(06/18/25 13:18:47.947:286) : avc:  denied  { net_admin } for  pid=1437 comm=iio-sensor-prox capability=net_admin  scontext=system_u:system_r:iiosensorproxy_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
----                                       
type=PROCTITLE msg=audit(06/18/25 13:18:47.947:287) : proctitle=/usr/libexec/iio-sensor-proxy 
type=SYSCALL msg=audit(06/18/25 13:18:47.947:287) : arch=x86_64 syscall=setsockopt success=yes exit=0 a0=0x7 a1=SOL_SOCKET a2=SO_ATTACH_FILTER a3=0x7ffe02a88930 items=0 ppid=1 pid=1437 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=iio-sensor-prox exe=/usr/libexec/iio-sensor-
proxy subj=system_u:system_r:iiosensorproxy_t:s0 key=(null)                           
type=AVC msg=audit(06/18/25 13:18:47.947:287) : avc:  denied  { sys_admin } for  pid=1437 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
type=AVC msg=audit(06/18/25 13:18:47.947:287) : avc:  denied  { sys_admin } for  pid=1437 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
type=AVC msg=audit(06/18/25 13:18:47.947:287) : avc:  denied  { sys_admin } for  pid=1437 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
type=AVC msg=audit(06/18/25 13:18:47.947:287) : avc:  denied  { sys_admin } for  pid=1437 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
type=AVC msg=audit(06/18/25 13:18:47.947:287) : avc:  denied  { sys_admin } for  pid=1437 comm=iio-sensor-prox capability=sys_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
----
```

*EDIT:* Do you need the backtrace as well ?

rfrohl commented:
But I suspect that the above is not helpful now, as I read further: [d4e89d2](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d4e89d212d401672e9cdfe825d947ee3a9fbe3f5) went into 6.12.29 as well.

I think I have no easy option to test without the commit. I could build a kernel without it, but that might take a bit of time. Could you let me know if that adds any value ? Would prefer to not waste time, if it is not needed.

rfrohl commented:
ok, was curious what will change and patch dependency was not as bad as I initially thought: 
Tested a custom 6.12.33 ([full auditing](https://fedoraproject.org/wiki/SELinux/Debugging#Enable_full_auditing) enabled , with 'semodule -DB' and both fixes) with three patches removed ([d4e89d](https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=d4e89d212d401672e9cdfe825d947ee3a9fbe3f5), [9f725e](https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=9f725eec8fc0b39bdc07dcc8897283c367c1a163) and [073fdb](https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=073fdbe02c69c43fb7c0d547ec265c7747d4a646); later two depend on d4e89d) :

```
----                                                                                  
type=PROCTITLE msg=audit(06/18/25 15:14:49.629:291) : proctitle=/usr/libexec/iio-sensor-proxy 
type=SYSCALL msg=audit(06/18/25 15:14:49.629:291) : arch=x86_64 syscall=setsockopt success=no exit=EPERM(Operation not permitted) a0=0x7 a1=SOL_SOCKET a2=SO_RCVBUFFORCE a3=0
x7ffecb6e4a84 items=0 ppid=1 pid=1442 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=iio-sensor-prox ex
e=/usr/libexec/iio-sensor-proxy subj=system_u:system_r:iiosensorproxy_t:s0 key=(null) 
type=AVC msg=audit(06/18/25 15:14:49.629:291) : avc:  denied  { net_admin } for  pid=1442 comm=iio-sensor-prox capability=net_admin  scontext=system_u:system_r:iiosensorprox
y_t:s0 tcontext=system_u:system_r:iiosensorproxy_t:s0 tclass=capability permissive=0  
----
```

HTH

zpytela commented:
kernel version may be misleading, we cannot compare directly kernels in different distros. As we cannot predict when (or even if) the capability check kernel change is merged, I am fine with the temporary dontaudit rule. What makes the "temporary" best is a one-line comment in the code, please do so. Also split the commit into two so that dontaudit can be reverted, and use wording+bz link like in d37dd6ab68 which seems to be wrong and should rather be reverted.

rfrohl committed:
Revert "Allow iio-sensor-proxy sendto to journald over a unix datagram socket"

This reverts commit d37dd6ab68cccd11a02ad48c026f8f85c78f6e46.

rfrohl committed:
Allow iio-sensor-proxy sendto to journald over a unix datagram socket

The permission actually allows iio-sensor-proxy send messages to kernel
unix datagram sockets as the journald socket is still labeled kernel_t.

The commit addresses the following AVC denial:
type=AVC msg=audit(..): avc:  denied  { sendto } for  pid=6002 comm="iio-sensor-prox" path="/systemd/journal/socket" scontext=system_u:system_r:iiosensorproxy_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=unix_dgram_socket permissive=0

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2368281
Fixes: bsc#1244577
Signed-off-by: Robert Frohl <rfrohl@suse.com>

rfrohl committed:
Temporary dontaudit iio-sensor-proxy sys_admin.

Relevant discussion can be found in
https://bugzilla.redhat.com/show_bug.cgi?id=2369326
as well as in
https://github.com/fedora-selinux/selinux-policy/pull/2734

Relevant kernel commit is d4e89d212d401672e9cdfe825d947ee3a9fbe3f5.

rfrohl commented:
I think the requested changes should be there, tried to include as much details in the temporary dontaudit commit as I have as well.

zpytela commented:
LGTM
